src/app.rs

```diff
@@ -1165,7 +1176,12 @@ impl SearchState {
     }
 
     fn is_string_match(&self, s: &str) -> bool {
-        s.contains(&self.query)
+        match self.whole_word {
+            true => Regex::new(format!("\\b{}\\b", self.query).as_str())
```

Comment:
I guess this also works (I was not sure whether `format!` would be happy with r-strings)
```suggestion
            true => Regex::new(format!(r"\b{}\b", self.query).as_str())
```

Reply:
I believe that r-strings work at a lexer level, so they're resolved before the `format!` macro ever sees them. Either way, I'd try it and if it compiles, may as well use it.

Reply:
Also, unrelated to the above: I think we need to escape the query string here, or else regex syntax in the user's string will be unintentionally executed. There's an API for that:

https://docs.rs/regex/latest/regex/fn.escape.html

Reply:
Sorry, one more thing. Instead of formatting the string and building a new Regex each time, we should really precompile the Regex and reuse it. That's a bit of extra state to track but I think it'll pay off in performance, especially since it'll reduce memory allocation dramatically.

Reply:
https://github.com/StanfordLegion/prof-viewer/pull/32/commits/2ae67336669857b6630952061bbdd5d680e62cf7 then https://github.com/StanfordLegion/prof-viewer/pull/32/commits/847f315fdc7b42d40abcd0e51bf9b45a872bce78

Caching is getting a bit involved, perhaps approaching time to factor out a separate object to manage caches that is amenable to unit tests?